src/note.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  SolidDataset, Thing,
+  getThing, createThing,
+  addUrl,
+  setInteger, setDecimal, setStringNoLocale, setBoolean,
+  getStringNoLocale, getInteger, getBoolean, getUrl,
+} from '@inrupt/solid-client';
+import { createNS } from "@ontologies/core"
+
+import {arrayToThings, thingsToArray} from "./collections"
+
+const noteNSUrl = "https://mysilio.garden/ontologies/note#"
+const noteNS = createNS(noteNSUrl)
+
+function addKeyValToThing(thing: Thing, key: string, value: any, path: number[]): Thing[] {
+  if (Array.isArray(value)) {
+    const [arrayThing, ...restOfThings] = arrayToThings(value, createThingFromSlateJSOElement, [...path, 0])
+    if (arrayThing) {
+      return [
+        addUrl(thing, key, arrayThing),
+        arrayThing,
+        ...restOfThings
+      ]
+    } else {
+      // TODO: is this the right thing to do? arrayThing should never be falsy so maybe throw an error?
+      return []
+    }
+  } else {
+
+    switch (typeof value) {
+      case 'string':
+        return [setStringNoLocale(thing, key, value)]
+      case 'boolean':
+        return [setBoolean(thing, key, value)]
+      case 'number':
+        if (Number.isInteger(value)) {
+          return [setInteger(thing, key, value)]
+        } else {
+          return [setDecimal(thing, key, value)]
+        }
+      default:
+        return [thing]
+    }
+  }
+
+}
+
+export function createThingFromSlateJSOElement(o: any, path: number[]): Thing[] {
+  const thing = createThing({ name: (o.id ? `${o.id}` : `el-${path.join("-")}`) })
+  const otherThings = Object.keys(o).reduce((m: Thing[], k: string) => {
+    const [mThing, ...mRest] = m;
+    const [thing, ...restOfThings] = addKeyValToThing(mThing, noteNS(k).value, o[k], path)
```

Comment:
It looks like all the places you use this in the downstream function are Inrupt fns. Those all actually take NamedNodes as well as URLStrings for the predicates, so you don't have to do `noteNS(k).value` here, you could just do `noteNS(k)` if you wanted.

Reply:
hm, I tried this and got:
<img width="985" alt="Screen Shot 2022-06-29 at 12 13 15 PM" src="https://user-images.githubusercontent.com/1113/176517482-a3f372cb-4eb5-4f31-b163-115217173277.png">

looks like addUrl doesn't accept a NamedNode?